index.js

```diff
@@ -157,6 +152,92 @@ app.get('/logout', (req, res) => {
     }
 });
 
+app.get('/dashboard', async (req, res) => {
+    if (!req.session.user || !req.cookies.user_sid) {
+        res.redirect('/login')
+        return
+    }
+
+    try {
+        let allLocations = await db.getLocations()
+        
+        let viewParams = {
+            locations: allLocations
+                .map(location => { 
+                    return { name: location.displayName, id: location.locationid }
+                })
+        }
+        viewParams.viewMessage = allLocations.length >= 1 ? 'Please select a location' : 'No locations to display'
+
+        res.send(Mustache.render(locationsDashboardTemplate, viewParams))
+    }
+    catch(err) {
+        console.log(err)
+        res.status(500).send()
+    }
+})
+
+app.get('/dashboard/:locationId', async (req, res) => {
+    if (!req.session.user || !req.cookies.user_sid) {
+        res.redirect('/login')
+        return
+    }
+    else if(typeof req.params.locationId !== 'string') {
+        let locations = await db.getLocations()
+        res.redirect('/dashboard/' + locations[0].id)
+        return
+    }
+
+    try {
+        let recentSessions = await db.getHistoryOfSessions(req.params.locationId)
```

Comment:
In Buttons we restrict this to only the 40 most-recent sessions. This list could end up being huge. Was it restricted in the old ODetect dashboard at all?

Reply:
It had a textbox to display x amount but Mario said that could be gotten rid of if I'm remembering right, I can put a cap on it pretty easily though if there's a number that makes sense to put it at, maybe @mariocimet and/or @sampathsatti have an opinion on that?

Reply:
We can try the last 40

Reply:
Okay, I'll throw that in there :)

Reply:
40 is fine, but sometimes I am guilty of quickly scanning the last 200 or so sessions to see what's up with a particular location

Reply:
Then let's make it 200 :smile:

Reply:
Done :)